input/dir/dir_input.go

```diff
@@ -0,0 +1,55 @@
+package dir
+
+import (
+	"crypto/sha512"
+	"hash"
+	"os"
+
+	"github.com/spacemonkeygo/errors"
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/input"
+	"polydawn.net/repeatr/lib/fshash"
+)
+
+const Type = "dir"
+
+// interface assertion
+var _ input.Input = &Input{}
```

Comment:
I like keeping them near every typedef.  It's the least confusing place to get the error emitted from.  (The Go style guide disagrees that you should even use this trick; IMO that's sensible if you're composing detachable packages, but utter nonsense if the interface is guaranteed to be in your imported packages anyway and it's critical to sanity.)